src/components/Sidebar/Menu.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { useLocation } from 'react-router';
+import { NavLink } from 'react-router-dom';
+import styled from 'styled-components';
+import { Flex, FlexProps } from '../Flex';
+import { Txt } from '../Txt';
+
+export interface MenuItem {
+	className?: string;
+	href: string;
+	title: string;
+	icon?: React.ReactNode;
+}
+
+const MenuItem = styled(Flex)<{ isCurrent: boolean }>`
+	height: 48px;
+	cursor: pointer;
+	border-bottom: 1px solid
+		${(props) =>
+			props.isCurrent
+				? props.theme.colors.quartenary.light
+				: props.theme.colors.tertiary.dark};
+	border-left: ${(props) =>
+		props.isCurrent ? `2px solid ${props.theme.colors.primary.main}` : 'none'};
+	background: ${(props) =>
+		props.isCurrent
+			? props.theme.colors.quartenary.light
+			: props.theme.colors.secondary.dark};
+
+	&:hover {
+		background: ${(props) =>
+			props.isCurrent
+				? props.theme.colors.quartenary.light
+				: props.theme.colors.tertiary.dark};
+	}
+`;
+
+export const Menu = ({
+	items,
+	isCollapsed,
+	...props
+}: {
+	items: MenuItem[];
+	isCollapsed: boolean;
+} & FlexProps) => {
+	const { pathname } = useLocation();
+
+	const currentItem = React.useMemo(() => {
+		return items.find((item) => pathname === item.href);
+	}, [items, pathname]);
+
+	return (
+		<Flex flexDirection="column" {...props}>
+			{items.map((item) => (
+				<NavLink key={item.href} to={item.href}>
+					<MenuItem
+						className={item.className}
+						width="100%"
+						isCurrent={currentItem === item}
```

Comment:
nit: consider a const at the beginning of the map for `const isCurrent = currentItem === item` since we reuse this 4 times